### PR TITLE
New abort sequence using condition variable

### PR DESCRIPTION
**Pull request by biscuut**

### CODE REVIEW

File: helpers.cpp [hunk 1]
```diff
@@ -323,7 +330,12 @@ void mach::SocketConn::input() {
     if (val == "op6") {
       poolptr->stop();
       std::cout<<"past stop" <<std::endl;
-      std::terminate();
+      std::terminate(); // ?
```
Comment (d-ibanescu) [inline helpers.cpp]: we can just kill this line it's unnecessary

File: CMakeLists.txt [hunk 2]
```diff
@@ -45,4 +45,4 @@ add_library(json INTERFACE)
 target_include_directories(json INTERFACE ${json_SOURCE_DIR}/include)
 
 # link all libs
-target_link_libraries(engine_computer PRIVATE ${LJM} ${BOOST_Libraries} json)
+target_link_libraries(engine_computer PRIVATE pthread ${LJM} ${BOOST_Libraries} json)
```
Comment (d-ibanescu) [inline CMakeLists.txt]: what is pthread linked for?

File: helpers.cpp (hunk 1 above)
Comment (CharelSuarez) [reply helpers.cpp:333]: oh yeah I was gonna ask how the abort sequence runs if we call std::terminate(), but if its unnecessary then no problemo

Comment (biscuut) [reply helpers.cpp]: oh yeah I was gonna ask how the abort sequence runs if we call std::terminate(), but if its unnecessary then no problemo

File: CMakeLists.txt (hunk 2 above)
Comment (biscuut) [reply CMakeLists.txt]: uhh the linking fails for me on WSL without it, is there something I'm missing? maybe updating ubuntu to the latest version?

Comment (d-ibanescu) [reply CMakeLists.txt]: oh i guess that's fine then, it's a gcc version moment probably

**Review by d-ibanescu — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (biscuut) [reply CMakeLists.txt]: probably 💀 I should probably just update wsl ubuntu instead of using the LTS version from 20 years ago

**Pull request merged**